cpu/power9/scom: working direct and first prototype of indirect SCOM access

IgorBagnucki opened the pull request.

Commit:
src/lib/prog_loaders.c: first prototype of scom indirect

Commit:
src/lib/prog_loaders.c: check hmer

krystian-hebel commented on src/lib/prog_loaders.c:
```diff
@@ -39,11 +40,167 @@ int prog_locate(struct prog *prog)
+		read_hmer() & SPR_HMER_XSCOM_STATUS);
+	for(int retries = 0; retries < XSCOM_IND_MAX_RETRIES; ++retries) {
+		printk(BIOS_EMERG, "READING addr = %llX\n", addr);
+		read_scom_direct(addr, &data);
```
This should be write

krystian-hebel commented on src/lib/prog_loaders.c:
```diff
@@ -39,11 +40,167 @@ int prog_locate(struct prog *prog)
+#define PPC_BIT(bit)		(0x8000000000000000UL >> (bit))
+#define PPC_BITMASK(bs,be)	((PPC_BIT(bs) - PPC_BIT(be)) | PPC_BIT(bs))
+#define XSCOM_ADDR_IND_FLAG		PPC_BIT(0)
+#define XSCOM_ADDR_IND_ADDR		PPC_BITMASK(12,31)
```
I think this should be (11,31), but it shouldn't cause a failure for this register

krystian-hebel commented on src/lib/prog_loaders.c:
```diff
@@ -39,11 +40,167 @@ int prog_locate(struct prog *prog)
+		read_hmer() & SPR_HMER_XSCOM_STATUS);
+	printk(BIOS_EMERG, "WRITING addr = %llX, data = %llX\n", addr, data);
+	write_scom_direct(addr, &data);
+	printk(BIOS_EMERG, "hmer = %llX, fail = %llX, done = %llX, status = %llX\n",
```
Maybe try to repeat whole `printk` a couple of times, this may be a timing issue: https://github.com/3mdeb/talos-hostboot/blob/master/src/usr/scom/scom.C#L489

Commit:
src/lib/prog_loaders.c:m Add debuging

krystian-hebel committed:
cpu/power9: move SCOM access functions to a separate file

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
cpu/power9: add directory to Makefile's subdirs

Also fix a typo and a build issue caused by incomplete type.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
Makefile.inc: add ECC for CBFS on POWER9

This may change after final PNOR layout is established. For now it eases
development and testing.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
tmp: move further stages higher in memory

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
arch/ppc64/bootblock_crt0.S: enable vector operations

Vector operations are sometimes generated for code such as
'uint8_t x[32] = {0}', this results in an exception when vector registers
(VR) are not enabled. VSR (vector-scalar register) operations are also
enabled, there is no reason not to.

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
